### PR TITLE
feat(resource): support update for subaccount_environment_instance

### LINKED ISSUE #100: Resource "subaccount_environment_instance" - Support Update Operation
Implementation of missing/not fully implemented UPDATE operation for the resource `subaccount_environment_instance`

### PR DESCRIPTION
**Pull request by kuntzed**
## Purpose
support update for subaccount_environment_instance resource

Also:
- add test utility for simple (not-)contains checks on resource attributes
- added TODO to get rid of copying over parameters from plan to state

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type

What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test the code

```
go test ./...
```

## What to Check

green test

## Other Information
Resolves #100

### CODE REVIEW

File: internal/provider/resource_subaccount_environment_instance.go [hunk 1]
```diff
@@ -260,20 +275,46 @@ func (rs *subaccountEnvironmentInstanceResource) Update(ctx context.Context, req
...
 
-	resp.Diagnostics.AddError("API Error Updating Resource Environment Instance (Subaccount)", "Update is not yet implemented.")
-
-	/* TODO: implementation of UPDATE operation
-	cliRes, err := gen.client.Execute(ctx, btpcli.Update, gen.command, plan)
+	_, _, err := rs.cli.Accounts.EnvironmentInstance.Update(ctx,
+		plan.SubaccountId.ValueString(),
+		plan.Id.ValueString(),
+		plan.PlanName.ValueString(),
+		plan.Parameters.ValueString(),
```
Comment (lechnerc77) [inline internal/provider/resource_subaccount_environment_instance.go]: To stay consistent with the CREATE scenario, would it make sense to embed the parameters in a struc (which would then also need to be adjusted in the facade)

**Review by lechnerc77 — COMMENTED** [review]
Pls. check the comment

File: internal/provider/resource_subaccount_environment_instance.go (hunk 1 above)
Comment (kuntzed) [reply internal/provider/resource_subaccount_environment_instance.go]: I was not able to observe a pattern when a struct is used. My impression is that it depends on the number of parameters. In this case, the create needs a lot more parameters and I was fine with having four of them in the signature of update. But if you prefer a struct, then I can change this.

Comment (lechnerc77) [reply internal/provider/resource_subaccount_environment_instance.go]: Indeed we are not consistent over the resources. However I would stay consistent within one resource and pass the parameters via a struc

Comment (kuntzed) [reply internal/provider/resource_subaccount_environment_instance.go]: Done

**Review by lechnerc77 — APPROVED** [review]
LGTM

**Pull request merged**